app/server/utils/ussd/ussd.py

```diff
@@ -0,0 +1,30 @@
+from typing import Optional
+
+from server import db
+from server.models.user import User
+from server.models.ussd import UssdMenu, UssdSession
+
+
+# TODO: in the future this should handle i18n more generally
+def menu_display_text_in_lang(current_menu: UssdMenu, user: Optional[User]) -> str:
+    if user is None or user.preferred_language is None:
+        return current_menu.display_text_en
+    else:
+        if user.preferred_language == "sw_KE":
+            return current_menu.display_text_sw
+        else:
+            return current_menu.display_text_en
+
+
+def create_or_update_session(session_id: str, user: User, current_menu: UssdMenu, user_input: str, service_code: str) -> UssdSession:
+    session = UssdSession.query.filter_by(session_id=session_id)
+    if session.count() > 0:
```

Comment:
originally I had it as `if session is None` and the test wouldn't hit this condition. is looking at count the correct way to test for existence of db entry?

Reply:
also, when i had it as `session.user_input = user_input` etc, that didn't seem to update correctly, but that was the syntax i saw elsewhere... was i missing something?

Reply:
`if session is None` is normally what i'd do. not sure why it's not working. I'll have a look at the tests on my local machine and see whats up. 

WRT the update process, I mean technically it works?

`session.update(...)` is a bulk update for all UssdSessions matching the filter `session_id=session_id`. Though presumably there should only be one session with that session_id?

I've only ever used `.update()` once: [https://github.com/teamsempo/SempoBlockchain/blob/7e9615a5a7c63bd893636bd632b61be03e03251b/eth_worker/sql_persistence/interface.py#L26](url)